src/popup/components/Modals/QrCodeReader.vue

```diff
@@ -34,137 +31,95 @@
-import BtnMain from '../buttons/BtnMain.vue';
-import QrScanIcon from '../../../icons/qr-scan.svg?vue-component';
-import IconBoxed from '../IconBoxed.vue';
+import AnimatedSpinner from '@/icons/animated-spinner.svg?skip-optimize';
+import Modal from '@/popup/components/Modal.vue';
+import BtnMain from '@/popup/components/buttons/BtnMain.vue';
+import IconBoxed from '@/popup/components/IconBoxed.vue';
+import QrScanIcon from '@/icons/qr-scan.svg?vue-component';
 
-export default {
+export default defineComponent({
   components: {
     Modal,
     BtnMain,
     IconBoxed,
+    AnimatedSpinner,
   },
   props: {
     title: { type: String, required: true },
     resolve: { type: Function, required: true },
     reject: { type: Function, required: true },
   },
-  setup() {
-    return {
-      QrScanIcon,
-    };
-  },
-  data: () => ({
-    mobile: IS_MOBILE_APP,
+  setup(props) {
     // allow camera while QRScanner is loading to not show cameraNotAllowed before actual check
-    cameraAllowed: IS_MOBILE_APP,
-    browserReader: null,
-    videoInputDevices: [],
-    headerText: '',
-  }),
-  watch: {
-    async cameraAllowed(value) {
-      if (!value) {
-        this.stopReading();
-        return;
-      }
+    const cameraStatus = ref<PermissionState>(
+      IS_MOBILE_APP ? 'granted' : 'denied',
+    );
+    const browserReader = ref<BrowserQRCodeReader | null>(null);
+    const browserReaderControls = ref<IScannerControls>();
+    const qrCodeVideo = ref<HTMLVideoElement>();
 
-      try {
-        this.resolve(await this.scan());
-      } catch (error) {
-        if (error.name === 'NotAllowedError') {
-          try {
-            await new Promise((resolve, reject) => {
-              if (IS_EXTENSION) {
-                openInNewWindow(
-                  browser.extension.getURL('./CameraRequestPermission.html'),
-                );
-                reject();
-              }
-              if (navigator.mediaDevices?.getUserMedia) {
-                navigator.mediaDevices.getUserMedia({ video: true }).then(resolve, reject);
-              } else reject(new Error('Sorry, your browser does not support getUserMedia'));
-            });
-          } catch {
-            this.cameraAllowed = false;
-          }
-          return;
-        }
-        handleUnknownError(error);
-      }
-    },
-    $route() {
-      this.resolve();
-    },
-  },
-  async mounted() {
-    if (this.mobile) {
-      if (await this.hasPermission()) {
-        await BarcodeScanner.prepare();
+    const route = useRoute();
+    const store = useStore();
+    const { t } = useI18n();
+
+    const cameraAllowed = computed(() => cameraStatus.value === 'granted');
+    const heading = computed(() => {
+      if (cameraStatus.value === 'granted') return t('modals.qrCodeReader.scanQr');
+      return t('modals.qrCodeReader.grantPermission');
+    });
+
+    function initBrowserReader() {
+      browserReader.value = new BrowserQRCodeReader();
+    }
```

Comment:
Previously the  reader library was dynamically imported when initiating. I think it was a good optimisation.

Reply:
We are initializing the `browserReader` `ref` with the type. One solution is to import `BrowserQRCodeReader` once as with a type-only import and then again dynamically if needed.